File: csharp/941.valid-mountain-array.cs
/*
 * @lc app=leetcode id=941 lang=csharp
 *
 * [941] Valid Mountain Array
 */

// @lc code=start
public class Solution {
    public bool ValidMountainArray(int[] A) {
        int n = A.Length, i;
        for (i = 0; i < n - 1 && A[i] < A[i + 1]; ++i) {}
        if (i == 0 || i == n - 1) {
            return false;
        }
        for (; i < n - 1 && A[i] > A[i + 1]; ++i) {}
        return i == n - 1;
    }
}
// @lc code=end

File: java/941.valid-mountain-array.java
/*
 * @lc app=leetcode id=941 lang=java
 *
 * [941] Valid Mountain Array
 */

// @lc code=start
class Solution {
    public boolean validMountainArray(int[] arr) {
        int n = arr.length;
        if (n < 3) {
            return false;
        }
        if (arr[1] <= arr[0]) {
            return false;
        }
        boolean dec = false;
        for (int i = 2; i < n; ++i) {
            if (dec) {
                if (arr[i] >= arr[i - 1]) {
                    return false;
                }
            } else {
                if (arr[i] == arr[i - 1]) {
                    return false;
                }
                dec = arr[i] < arr[i - 1];
            }
        }
        return dec;
    }
}
// @lc code=end

